app/models/service_template.rb

```diff
@@ -340,49 +353,97 @@ def provision_action
     resource_actions.find_by(:action => "Provision")
   end
 
+  def update_resource_actions(ae_endpoints)
+    resource_action_list.each do |action|
+      resource_params = ae_endpoints[action[:param_key]]
+      resource_action = resource_actions.find_by(:action => action[:name])
+      # If the action exists in updated parameters
+      if resource_params
+        # And the resource action exists on the template already, update it
+        if resource_action
+          resource_action.update_attributes!(resource_params)
```

Comment:
We should slice selective attributes known to `ResourceAction` just like how to create resource actions.